qiskit/aqua/operators/primitive_ops/summed_pauli_op.py

```diff
@@ -0,0 +1,362 @@
+                    z_factor = np.product(
+                        1 - 2 * np.logical_and(bitstr, corrected_z_bits), axis=1
+                    )
+                    y_factor = np.product(
+                        np.sqrt(
+                            1
+                            - 2 * np.logical_and(corrected_x_bits, corrected_z_bits)
+                            + 0j
+                        ),
+                        axis=1,
+                    )
+                    for i, n_str in enumerate(new_str):
+                        new_dict[n_str] = (
+                            v * z_factor[i] * y_factor[i] * coeffs[i]
+                        ) + new_dict.get(n_str, 0)
+                    return DictStateFn(new_dict, coeff=self.coeff * front.coeff)
+
+            elif isinstance(front, StateFn) and front.is_measurement:
+                raise ValueError("Operator composed with a measurement is undefined.")
+
+            # Composable types with PauliOp
+            elif isinstance(front, (SummedPauliOp, PauliOp, CircuitOp, CircuitStateFn)):
+                return self.compose(front)
+
+            # Covers VectorStateFn and OperatorStateFn
+            elif isinstance(front, OperatorBase):
+                return self.to_matrix_op().eval(front.to_matrix_op())  # type: ignore
+        return None
+
+    def exp_i(self) -> OperatorBase:
+        """ Return a ``CircuitOp`` equivalent to e^-iH for this operator H. """
+        # TODO: optimize for some special cases
+        from ..evolutions.evolved_op import EvolvedOp
+
+        return EvolvedOp(self)
+
+    def to_instruction(self) -> Instruction:
+        return self.to_matrix_op().to_circuit().to_instruction()  # type: ignore
+
+    def to_pauli_op(self, massive: bool = False) -> OperatorBase:
+        from .pauli_op import PauliOp
+
+        def to_real(x):
+            return x.real if np.isreal(x) else x
+
+        def to_native(x):
+            return x.item() if isinstance(x, np.generic) else x
+
+        if len(self.primitive) == 1:
+            return PauliOp(
+                Pauli(x=self.primitive.table.X[0], z=self.primitive.table.Z[0]),  # type: ignore
+                to_native(to_real(self.primitive.coeffs[0])) * self.coeff,  # type: ignore
+            )
+        return SummedOp(
+            [
+                PauliOp(
+                    Pauli(x=s.table.X[0], z=s.table.Z[0]),
+                    to_native(to_real(s.coeffs[0])),
+                )
+                for s in self.primitive
+            ],
+            coeff=self.coeff,
+        )
+
+    def __getitem__(self, offset: Union[int, slice]) -> "SummedPauliOp":
+        """Allows array-indexing style access to the ``SummedPauliOp``.
+
+        Args:
+            offset: The index of ``SummedPauliOp``.
+
+        Returns:
+            The ``SummedPauliOp`` at index ``offset``,
+        """
+        return SummedPauliOp(self.primitive[offset], self.coeff)
+
+    def __len__(self) -> int:
```

Comment:
I will note that for WeightedPaulitOperator, if we ended up with no paulis in the list, then there was an is_empty() method. I imagine that the table can be empty here too and that len would then be 0?

Reply:
We can meke empty operator (like SummedPauliOp(SparsePauliOp([]))), but its len is `1`.
If necessary, I will make is_empty method or utility function for this purpose.

Reply:
Is this because by default it logically contains Identity? If my table was created from a single pauli, say ("XYXX", 0.67), is the len now 2? I guess if all the coeffs are below the atol when I do a reduce, such that they are all removed, then we end up with len=1? This was mainly in chemistry where we had this that when building out and manipulating the FermionicOperator and finally creating a WeightedPauliOperator it would end up being empty. We passed these on since we had to take care of such an event in VQE anyway if a user directly passed in an empty operator. We just skipped these when computing an expectation value using the var form circuits and just simply returned 0 as the value for these  later. When it came to opflow it ended up more complicated for some reason https://github.com/Qiskit/qiskit-aqua/blob/1524d09319829f9dc66cf602bec540ba04d9959f/qiskit/aqua/algorithms/minimum_eigen_solvers/vqe.py#L254-L266 as perhaps there was not awareness this happened. Hence my commenting on this aspect

Reply:
For example, 
```python
h2 = SparsePauliOp.from_list(
            [
                ("II", -1.052373245772859),
                ("IZ", 0.39793742484318045),
                ("ZI", -0.39793742484318045),
                ("ZZ", -0.01128010425623538),
                ("XX", 0.18093119978423156),
            ]
        )
(1e-10*h2).simplify()
```
outputs
```
SparsePauliOp([[False, False, False, False]],
              coeffs=[0.+0.j])
```

To check empty operator, it seems enough just to check that coefficient are zero.